aas-web-ui/src/components/SubmodelPlugins/PCF.vue

```diff
@@ -0,0 +1,173 @@
+        return {
+            PCFCO2eq: [] as Array<any>,
+            TCFCO2eq: [] as Array<any>,
+        };
+    },
+    mounted() {
+        this.initializeProductCarbonFootprint();
+        this.initializeTransportCarbonFootprint();
+    },
+    computed: {
+        SelectedNode() {
+            return this.aasStore.getSelectedNode;
+        },
+        colorPercentage() {
+            return this.calculatePercentage(this.PCFCO2eqTotal, this.TCFCO2eqTotal);
+        },
+        pcfValueColor() {
+            return this.getThemeColor('pcf');
+        },
+        tcfValueColor() {
+            return this.getThemeColor('tcf');
+        },
+        fingersColor() {
+            return this.getThemeColor('fingers');
+        },
+        PCFCO2eqTotal(): number {
+            return this.PCFCO2eq.reduce((acc, item) => acc + item.value, 0);
+        },
+        TCFCO2eqTotal(): number {
+            return this.TCFCO2eq.reduce((acc, item) => acc + item.value, 0);
+        }
+    },
+    methods: {
+        calculatePercentage(productValue: number, transportValue: number): number {
+            const total = productValue + transportValue;
+            return total ? (productValue / total) * 100 : 0;
+        },
+        getThemeColor(colorKey: string): string {
+            const isDark = this.$vuetify.theme.global.current.dark;
+            const themeColors = isDark ? this.$vuetify.theme.themes.dark.colors : this.$vuetify.theme.themes.light.colors;
+            return themeColors[colorKey];
+        },
+        initializeCarbonFootprint(semanticId: string, value: 'PCFCO2eq' | 'TCFCO2eq') {
+            // console.log('Component Mounted. SubmodelElementData:', this.submodelElementData);
+            if (Object.keys(this.submodelElementData).length === 0) {
+                return;
+            }
+            try {
+                const carbonFootprintData = { ...this.submodelElementData };
+                const entries = carbonFootprintData.submodelElements.filter((elem: any) => {
+                    return elem.semanticId?.type === 'ExternalReference' &&
+                        elem.semanticId?.keys?.[0]?.type === 'GlobalReference' &&
+                        elem.semanticId?.keys?.[0]?.value === semanticId;
+                });
+                if (entries.length === 0) {
+                    console.warn(`No entries found for ${semanticId}.`);
+                    return;
+                }
+                this[value] = entries.map((entry: any) => ({
+                    semanticId: semanticId,
+                    value: parseFloat(entry.value?.find((val: any) => val?.idShort?.includes('CO2eq'))?.value) || 0
+                }));
+            } catch (error) {
+                console.error(`Error initializing ${semanticId}:`, error);
+            }
+        },
+        initializeProductCarbonFootprint() {
+            this.initializeCarbonFootprint('https://adminshell.io/idta/CarbonFootprint/ProductCarbonFootprint/0/9', 'PCFCO2eq');
```

Comment:
Please correct IRI to "https://admin-shell.io/idta/CarbonFootprint/ProductCarbonFootprint/0/9" (missing "-" in "adminshell" in your IRI)